### PR TITLE
migrate to mdoc

### PR DESCRIPTION
**Pull request by LukaJCB**
Should address #56

### CODE REVIEW

**Comment by LukaJCB** [comment]
For one reason or another `mdoc` hangs on this file: modules/docs/src/main/tut/docs/index.md

File: build.sbt [hunk 1]
```diff
@@ -132,6 +133,8 @@ val docs = project
     micrositeGithubRepo := "cats-retry",
     micrositeGitterChannel := true,
     micrositeTwitterCreator := "@cbirchall",
+    micrositeCompilingDocsTool := WithMdoc,
+    mdocIn := (sourceDirectory in Compile).value / "tut",
```
Comment (cb372) [inline build.sbt]: If you manage to get this working, let's rename `src/tut` to `src/mdoc` to avoid confusion.

**Comment by cb372** [comment]
Interesting. Is it worth opening an issue on mdoc for that?

**Comment by LukaJCB** [comment]
This is ready for re-review :)

**Comment by cb372** [comment]
Woah, you just closed and re-opened it in front of my eyes.

I wonder if GitHub got confused because the PR was created before we set up Actions?

Currently the `docs` action is configured to run `sbt docs/tut`. Could you update it to compile the docs with mdoc please. Not sure what the sbt task name is.

**Comment by LukaJCB** [comment]
Done :)

**Comment by cb372** [comment]
Looks good! But I ran it locally and spotted one formatting bug:

Screenshot 2019-11-05 at 16 44 08

About GH actions, it seems like they somehow got borked on this repo. I disabled the feature and re-enabled it and it started working again on this PR: #110. Will be interesting to see what happens if you add another commit.

**Pull request merged**